plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -535,12 +534,17 @@ public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
     }
 
     @VisibleForTesting
-    static JdbcColumnHandle createSyntheticJoinProjectionColumn(JdbcColumnHandle column, int nextSyntheticColumnId)
+    static JdbcColumnHandle createSyntheticJoinProjectionColumn(JdbcColumnHandle column, int nextSyntheticColumnId, OptionalInt optionalMaxColumnNameLength)
     {
         verify(nextSyntheticColumnId >= 0, "nextSyntheticColumnId rolled over and is not monotonically increasing any more");
```

Comment:
`orElse` / `orElseGet` can only return an object of type `T` for `Optional<T>` which is not this case.

There is `ifPresent` / `ifPresentOrElse` which leads to not more readable code because of a very large lambda + atomicreference needing to be used since lambdas need assignments to be effectively final:

```
    @VisibleForTesting
    static JdbcColumnHandle createSyntheticJoinProjectionColumn(JdbcColumnHandle column, int nextSyntheticColumnId, OptionalInt optionalMaxColumnNameLength)
    {
        verify(nextSyntheticColumnId >= 0, "nextSyntheticColumnId rolled over and is not monotonically increasing any more");

        final String separator = "_";
        AtomicReference<JdbcColumnHandle> ret = new AtomicReference<>();
        optionalMaxColumnNameLength.ifPresentOrElse(maxColumnNameLength -> {
                    int nextSyntheticColumnIdLength = String.valueOf(nextSyntheticColumnId).length();
                    verify(maxColumnNameLength >= nextSyntheticColumnIdLength, "Maximum allowed column name length is %s but next synthetic id has length %s", maxColumnNameLength, nextSyntheticColumnIdLength);

                    if (nextSyntheticColumnIdLength == maxColumnNameLength) {
                        ret.set(JdbcColumnHandle.builderFrom(column)
                                .setColumnName(String.valueOf(nextSyntheticColumnId))
                                .build());
                    }

                    if (nextSyntheticColumnIdLength + separator.length() == maxColumnNameLength) {
                        ret.set(JdbcColumnHandle.builderFrom(column)
                                .setColumnName(separator + nextSyntheticColumnId)
                                .build());
                    }

                    // fixedLength only accepts values > 0, so the cases where the value would be <= 0 are handled above explicitly
                    String truncatedColumnName = fixedLength(maxColumnNameLength - separator.length() - nextSyntheticColumnIdLength)
                            .split(column.getColumnName())
                            .iterator()
                            .next();
                    ret.set(JdbcColumnHandle.builderFrom(column)
                            .setColumnName(truncatedColumnName + separator + nextSyntheticColumnId)
                            .build());
                },
                () -> JdbcColumnHandle.builderFrom(column)
                        .setColumnName(column.getColumnName() + separator + nextSyntheticColumnId)
                        .build());
        return ret.get();
    }
```